common/src/main/java/org/apache/rocketmq/common/config/AbstractRocksDBStorage.java

```diff
@@ -386,7 +386,9 @@ public synchronized boolean shutdown() {
             }
             //4. close db.
             if (db != null) {
-                this.db.syncWal();
+                if (!this.readOnly) {
+                    this.db.syncWal();
+                }
```

Comment:
It would be better to integrate two if statements into one

Reply:
Good catch. Let me take a look.